tests/functional/sse-kms-migration/helpers.js

```diff
@@ -14,20 +37,121 @@ function getKey(key) {
     return config.kmsHideScalityArn ? getKeyIdFromArn(key) : key;
 }
 
-// for Integration use default profile, in cloudserver use vault profile
+// For Integration use default profile, in cloudserver use vault profile
 const credsProfile = process.env.S3_END_TO_END === 'true' ? 'default' : 'vault';
-const s3config = getConfig(credsProfile, { signatureVersion: 'v4' });
-const s3 = new S3(s3config);
+
+// Create custom agents with specific pooling settings
+const httpAgent = new HttpAgent({
+    keepAlive: true,
+    keepAliveMsecs: 30000,
+    maxSockets: 50,
+    maxFreeSockets: 10,
+    timeout: 120000,
+});
+
+const httpsAgent = new HttpsAgent({
+    keepAlive: true,
+    keepAliveMsecs: 30000,
+    maxSockets: 50,
+    maxFreeSockets: 10,
+    timeout: 120000, 
+});
+
+const s3config = {
+    ...getConfig(credsProfile, {}),
+    requestHandler: new NodeHttpHandler({
+        connectionTimeout: 120000,
+        socketTimeout: 120000,
+        httpAgent,
+        httpsAgent,
+    }),
+    maxAttempts: 8,
```

Comment:
Usually we can keep 3, this should be enough (and one could argue it should be 1 for a test suite, as if we hide transient errors, we might miss real bugs; and in such case we could just remove the retry logic)